### ISSUE TITLE
`np.nditer` should possibly allow access of `operands` after with statement `__exit__`

### ISSUE THREAD

**Issue by seberg** [numpy/numpy#12836]
I have not thought it out, but there was a (now cleared out) issue about `np.nditer` not allowing to access the `it.operands` after the `with` statement is finished. For return operands, accessing it only after finishing everything doesn't seem unreasonable though.

**Comment by mattip** [comment]
Duplicate of #10956. It is complicated since the exact meaning of `it.operands` when read-write is enabled is not clear. Does it mean the input operand, the temporary buffered scratch work area being updated with new values during iteration, or the output array with a possible different incompatible `dtype` from the input. In #10956 it was suggested that we create a new `outputs` attribute with a clear meaning, and allow that to be returned.

**Comment by seberg** [comment]
Ah, sorry did not see that one, so will close. `outputs` sounds like a pretty reasonable choice. On a completely different thing, since I was silenced ;). Were the NEPs hanging because of the writing process, or because of discussions around metaclass and instances being type's?
